sdk/Lusid.Sdk.Tests/TokenProviderTests.cs

```diff
@@ -101,5 +92,24 @@ public void CanUseTokenProviderConfiguration()
             var __ = config.AccessToken;
             mockTokenProvider.Verify(x => x.GetAuthenticationTokenAsync(), Times.Exactly(2));
         }
+
+        [Test]
+        public void Can_Retrieve_Same_Token_From_Multiple_Threads()
+        {
+            const int threadCount = 100;
+            
+            //    create threads with calls to get a token
+            var providers = Enumerable.Repeat(new ClientCredentialsFlowTokenProvider(ApiConfig.Value), threadCount).ToList();
+            var requests = providers.Select(p => p.GetAuthenticationTokenAsync());
+
+            //    get the tokens
+            Task.WaitAll(requests.ToArray());
+
+            //    all requests must have the same token i.e. reuse a valid token
+            var token = requests.First().Result;
+            var allEqual = requests.All(r => r.Result == token);
+            
+            Assert.That(allEqual, Is.True, "Requests have different tokens");
```

Comment:
requests.Distinct(r => r.Result).Count(), Is.EqualTo(1) ?

Reply:
Good shout